cmd/common/image/inspect.go

```diff
@@ -107,35 +109,96 @@ func NewInspectCmd(runtime runtime.Runtime) *cobra.Command {
 				"content":     string(v.Content),
 			}
 		}
+
+		var customResult interface{}
+		extraInfo, _ := cmd.PersistentFlags().GetString("extra-info")
+		if extraInfo != "" {
+			if _, ok := info.ExtraInfo.Data[extraInfo]; !ok {
+				return fmt.Errorf("extra info %q not found", extraInfo)
+			}
+
+			jsonPath, _ := cmd.PersistentFlags().GetString("jsonpath")
+			dataEntry := info.ExtraInfo.Data[extraInfo]
+			if dataEntry.ContentType == "application/json" {
+				// unmarshal the JSON content
+				if err := json.Unmarshal(dataEntry.Content, &customResult); err != nil {
+					return fmt.Errorf("unmarshalling JSON content: %w", err)
+				}
+			} else {
+				customResult = string(dataEntry.Content)
+			}
+			if jsonPath != "" {
+				// unmarshal the JSON content
+				var dataEntryContent interface{}
+				if err := json.Unmarshal(dataEntry.Content, &dataEntryContent); err != nil {
```

Comment:
`json.Unmarshal(dataEntry.Content, ...)` is potentially done twice: line 124 and 133. The previous unmarshalling should be reused if already done.

To simplify the code, `--jsonpath` could be forbidden if the content is not json in the first place. At the moment, it errors out if I try to use `--jsonpath` on something that is not json:
```
$ sudo ig image inspect ghcr.io/inspektor-gadget/gadget/trace_tcp --extra-info 'ebpf.sequence' -o yaml --jsonpath='[*]'
Error: unmarshalling JSON content: invalid character 's' looking for beginning of value
```